### PR TITLE
Changelog for 18.2

### PR DESCRIPTION
**Pull request by gnoff**
Changelog for 18.2

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,30 @@
+## 18.2.0 (TODO)
+
+### React
+
+* Only treat updates to Suspended components as a new mount in Legacy Mode ([@rickhanlonii](https://github.com/rickhanlonii)) in [#24530](https://github.com/facebook/react/pull/24530)
```
Comment (gaearon) [inline CHANGELOG.md:5]: We don't use the term "legacy mode" in public communication much. Also this doesn't have any observable behavior (I think), it's just an internal refactor. So let's omit it?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
+* Fix default `highWaterMark` behavior of `ReadableStream` in `renderToReadableStream` ([@jplhomer](https://github.com/jplhomer)) in [#24641](https://github.com/facebook/react/pull/24641)
+
+### Server Components (Experimental)
+
+* **useId Hook** `useId` is now supported in Server Components. useId's uniqueness guarantees are different in Server Components because there is no client hydration but there can be ID collisions across refetches if you refetch only a portion of the application on subsequent requests. To avoid collisions `useId` will use an option `identifierPrefix` which needs to be chose to ensure uniqueness across multiple requests. An additional nuanced behavior is that if you pass a Server Component using `useId` to a Client Component and that client component renders the passed in element into more than one place, the id's within the Server Component elements will not be unique. In the future we may add a warning in Dev if you render a Server Component that uses `useId` into more than one place on the client. ([@gnoff](https://github.com/gnoff)) in [#24172](https://github.com/facebook/react/pull/24172)
```
Comment (gaearon) [inline CHANGELOG.md:26]: Since Server Components are experimental it's enough for this description to live in PR. Whereas the changelog entry could be just like

- Add support for `useId()` inside Server Components.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+* Support `hydrateRoot` hydrating into `Document` as the `container` ([@gnoff](https://github.com/gnoff)) in [#24523](https://github.com/facebook/react/pull/24523)
+* Fix extraneous hydration errors when something suspends or errors inside a Suspense boundary ([@gnoff](https://github.com/gnoff)) in [#24480](https://github.com/facebook/react/pull/24480)
+* Fix producing extraneous recoverable errors when an inner Suspense boundary suspends while an outer Suspense boundary hydrates successfully ([@acdlite](https://github.com/acdlite)) in [#24532](https://github.com/facebook/react/pull/24532) and ([@gnoff](https://github.com/gnoff)) in [#24618](https://github.com/facebook/react/pull/24618)
+* Fix ignored setState in Safari when microtasks are flushed too early due to browser bug ([@gaearon](https://github.com/gaearon)) in [#24459](https://github.com/facebook/react/pull/24459) 
+* Do not allow `createPortal` to accept `Document` as the `container` ([@sebmarkbage](https://github.com/sebmarkbage)) in [#24496](https://github.com/facebook/react/pull/24496)
```
Comment (gaearon) [inline CHANGELOG.md:14]: This is a types-only Flow change so it shouldn't be in the changelog

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+### React
+
+* Only treat updates to Suspended components as a new mount in Legacy Mode ([@rickhanlonii](https://github.com/rickhanlonii)) in [#24530](https://github.com/facebook/react/pull/24530)
+
+### React DOM
+
+* Provide a Component Stack in Dev to `onRecoverableError` through an `errorInfo` second argument, similar to `componentDidCatch` in Error Boundaries ([@gnoff](https://github.com/gnoff)) in [#24591](https://github.com/facebook/react/pull/24591)
+* Support `hydrateRoot` hydrating into `Document` as the `container` ([@gnoff](https://github.com/gnoff)) in [#24523](https://github.com/facebook/react/pull/24523)
+* Fix extraneous hydration errors when something suspends or errors inside a Suspense boundary ([@gnoff](https://github.com/gnoff)) in [#24480](https://github.com/facebook/react/pull/24480)
+* Fix producing extraneous recoverable errors when an inner Suspense boundary suspends while an outer Suspense boundary hydrates successfully ([@acdlite](https://github.com/acdlite)) in [#24532](https://github.com/facebook/react/pull/24532) and ([@gnoff](https://github.com/gnoff)) in [#24618](https://github.com/facebook/react/pull/24618)
```
Comment (gaearon) [inline CHANGELOG.md:12]: I would unify these two items as "Fix false positive hydration errors with Suspense"

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,30 @@
+## 18.2.0 (TODO)
+
+### React
+
+* Only treat updates to Suspended components as a new mount in Legacy Mode ([@rickhanlonii](https://github.com/rickhanlonii)) in [#24530](https://github.com/facebook/react/pull/24530)
+
+### React DOM
+
+* Provide a Component Stack in Dev to `onRecoverableError` through an `errorInfo` second argument, similar to `componentDidCatch` in Error Boundaries ([@gnoff](https://github.com/gnoff)) in [#24591](https://github.com/facebook/react/pull/24591)
+* Support `hydrateRoot` hydrating into `Document` as the `container` ([@gnoff](https://github.com/gnoff)) in [#24523](https://github.com/facebook/react/pull/24523)
```
Comment (gaearon) [inline CHANGELOG.md:10]: This was already supported, the bug was that it couldn't cleanly client-render into it. I'd maybe rephrase as "Fix hydrating into document causing a blank page on mismatch".

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,30 @@
+## 18.2.0 (TODO)
+
+### React
+
+* Only treat updates to Suspended components as a new mount in Legacy Mode ([@rickhanlonii](https://github.com/rickhanlonii)) in [#24530](https://github.com/facebook/react/pull/24530)
+
+### React DOM
+
+* Provide a Component Stack in Dev to `onRecoverableError` through an `errorInfo` second argument, similar to `componentDidCatch` in Error Boundaries ([@gnoff](https://github.com/gnoff)) in [#24591](https://github.com/facebook/react/pull/24591)
```
Comment (gaearon) [inline CHANGELOG.md:9]: We avoid abbreviations like "in Dev" in changelog, let's say "in development". I'd maybe shorten as

- Provide a component stack as a second argument to `onRecoverableError`.

**Comment by gnoff** [comment]
@gaearon pushed an update with your recommended changes

**Comment by gaearon** [comment]
thanks! i pushed a few small changes (mostly formatting fixes and simplifying the wording)

**Comment by gnoff** [comment]
Appreciate it, i'll get the hang of the communication style soon :)
